src/cascadia/TerminalApp/CommandPalette.xaml

```diff
@@ -35,220 +35,194 @@ the MIT License. See LICENSE in the project root for license information. -->
+                        <!-- command label -->
+                        <ColumnDefinition Width="*"/>
+                        <!-- key chord -->
+                        <ColumnDefinition Width="16"/>
+                        <!-- gutter for scrollbar -->
+                    </Grid.ColumnDefinitions>
+
+                    <IconSourceElement
+                                Grid.Column="0"
+                                Width="16"
+                                Height="16"
+                                IconSource="{x:Bind Item.Icon,
+                                                    Mode=OneWay,
+                                                    Converter={StaticResource IconSourceConverter}}"/>
+
+                    <local:HighlightedTextControl
+                                Grid.Column="1"
+                                HorizontalAlignment="Left"
+                                Text="{x:Bind HighlightedName, Mode=OneWay}"/>
+
+                    <!-- The block for the key chord is only visible
+                            when there's actual text set as the label. See
+                            CommandKeyChordVisibilityConverter for details.
+                            We're setting the accessibility view on the
+                            border and text block to Raw because otherwise,
+                            Narrator will read out the key chord. Problem is,
+                            it already did that because it was the list item's
+                            "AcceleratorKey". It's redundant. -->
+                    <Border
+                                Grid.Column="2"
+                                Visibility="{x:Bind Item.KeyChordText,
+                                                Mode=OneWay,
+                                                Converter={StaticResource CommandKeyChordVisibilityConverter}}"
+                                Style="{ThemeResource KeyChordBorderStyle}"
+                                Padding="2,0,2,0"
+                                HorizontalAlignment="Right"
+                                VerticalAlignment="Center"
+                                AutomationProperties.AccessibilityView="Raw">
+
+                        <TextBlock
+                                    Style="{ThemeResource KeyChordTextBlockStyle}"
+                                    FontSize="12"
+                                    Text="{x:Bind Item.KeyChordText, Mode=OneWay}"
+                                    AutomationProperties.AccessibilityView="Raw" />
+                    </Border>
+                </Grid>
             </DataTemplate>
 
             <DataTemplate x:Key="NestedItemTemplate" x:DataType="local:FilteredCommand">
+                <Grid HorizontalAlignment="Stretch" ColumnSpacing="8" AutomationProperties.Name="{x:Bind Item.Name, Mode=OneWay}" AutomationProperties.AcceleratorKey="{x:Bind Item.KeyChordText, Mode=OneWay}">
```

Comment:
Are we losing the HelpText (More options) from `x:Uid="CommandPalette_MoreOptions"`

Reply:
Yeah -- we gotta keep this. I'll block for the answer. :smile:

Reply:
I'm a bit worried we're losing a couple things by leaving ListViewItem behind. Hmm. Can the ChoosingXyz event _not_ return a ListViewItem?

Reply:
Can you explain what do you mean by "leaving ListViewItem behind"? We do create it but in the event handler and not here.

ChoosingItemContainer must choose a container :smile: or more precisely - SelectorItem, which is a parent for *ViewItem. 

What do you believe will be broken? How can I test that? I mean I played with Narrator and it felt OK.. but probably I missed stuff.

Reply:
I mean I lost MoreOptions (I was not aware of it unfortunately). I guess I can try to fix this :blush: But are there additional concerns?

Reply:
My main concern was the MoreOptions call-out. Because we had some customizations on each ListViewItem, and I was not certain if they would be transferred to the Grid that we're using instead.

Because Grid is a Panel, it cannot have focus and it cannot have a number of properties on it.. but ListViewItem is a Control, and it can receive focus and stuff like that. Just wanted to make sure we were making the transition with intention. :smile:

Reply:
Like: I don't know what it means to have an AcceleratorKey on a Grid. If it works fine in narrator (that is: it reads out the accelerator!), then it's fine.

Reply:
Yeah. I see your point. I am somewhat worried as well, that is why I tested it with Narrator, which did read the command names and stuff like that, but I don't know if it is the right way to implement... and also if it says everything that should be said. 
I will try to play with 1.6 to compare. 
If Carlos could check, this could probably help as well.

Reply:
I will fix this by binding the automation properties in the code behind, and resubmit. Sorry.

Reply:
We're getting to the things I hate about XAML. It's like... we did all this work to get _away_ from manually setting up specific values in the codebehind _or in the xaml document_ for specific types of items, right? It was good and clean and smart: the DataTemplate provided everything we needed for the data type. And here, we're forced to add detection to ChoosingItemContainer (sp) to set up the container correctly based on the data type. Something the data template should be able to handle. All because of a silly issue. :smmile:

Reply:
I have expressed my distaste in the linked Xaml issue. Thanks for debugging through WinUI so much. 😀 It can be enough to drive a person mad.

Reply:
Okay, one last e-mail. It does not work if we add `x:Uid="CommandPalette_MoreOptions"` to the Grid?

Reply:
No, it doesn't. There is no AutomationPeer for the grid. I am still wondering why the text is read correctly. Most probably due to the default automation peer on the internal text block.